### PR TITLE
Python: Update sessions python plugin to work with the correct endpoints

### LINKED ISSUE #6934: Python: Bug:  ‎Python's SessionsPythonTool is using wrong REST data and endpoints for Azure Container Apps Sessions
**Describe the bug**
The request and response bodies for interacting with the ACA serverless code interpreter sessions is incorrect.

**To Reproduce**
1. Go to the sample workbook ['azure_python_code_interpreter_function_calling'](https://github.com/microsoft/semantic-kernel/blob/8fc645d1fe152a53420db484ed85afc7cce903cb/python/samples/concepts/auto_function_calling/azure_python_code_interpreter_function_calling.py)
2. Try using the file upload and/or code execution functions
3. See errors

**Expected behavior**
To be able to execute code interpreter sessions in ACA

**Platform**
 - OS: Windows
 - IDE: Visual Studio, VS Code
 - Language: Python
 - Source: main branch

**Additional context**
REST API specifications: https://learn.microsoft.com/en-us/azure/container-apps/sessions-code-interpreter

For example:
The File Upload API endpoint should be `files/upload` not `python/uploadFile`. It also returns a list of files, rather than a single file metadata. method: https://github.com/microsoft/semantic-kernel/blob/8fc645d1fe152a53420db484ed85afc7cce903cb/python/semantic_kernel/core_plugins/sessions_python_tool/sessions_python_plugin.py#L177

I believe all endpoints and data are incorrect, but haven't tested the all of the plugin's functions.

### PR DESCRIPTION
**Pull request by moonbox3**
### Motivation and Context

The Sessions Python Plugin was working properly, but some users reported that the endpoints needed updating. We had made previous updates to the plugin, but they didn't account for the update API endpoints and request body changes shown [here](https://learn.microsoft.com/en-us/azure/container-apps/sessions-code-interpreter).

### Description

This PR updates the plugin tool to work with the documented endpoints and request bodies. The unit tests were also updated. The concept sample for the Azure Python sessions plugin was tested with all methods to: execute code, upload multiple files, list files, and download them.
- Additionally, improve error handling for each of the function calls so the proper error message is surfaced to the user via SK exceptions.
- Fix path building and making sure the sessions endpoint is converted to a string from the Pydantic https type.
- Closes #6934 

### Contribution Checklist

- [X] The code builds clean without any errors or warnings
- [X] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [X] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

**Comment by markwallace-microsoft** [comment]
<a href="https://github.com/microsoft/semantic-kernel/blob/be521ffbb8b244612b2762905323893989de8743/README.md">Py3.10 Test Coverage</a><details><summary>Python 3.10 Test Coverage Report • </summary><table><tr><th>File</th><th>Stmts</th><th>Miss</th><th>Cover</th><th>Missing</th></tr><tbody><tr><td><b>TOTAL</b></td><td><b>6499</b></td><td><b>753</b></td><td><b>88%</b></td><td>&nbsp;</td></tr></tbody></table><i>report-only-changed-files is enabled. No files were changed during this commit :)</i></details>

## Python 3.10 Unit Test Overview
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 1558 | 1 :zzz: | 0 :x: | 0 :fire: | 23.965s :stopwatch: |

File: python/semantic_kernel/core_plugins/sessions_python_tool/sessions_python_plugin.py [hunk 1]
```diff
@@ -159,19 +163,24 @@ async def execute_code(self, code: Annotated[str, "The valid Python code to exec
...
+                json=request_body,
+            )
+            response.raise_for_status()
+            result = response.json()["properties"]
+            return f"Result:\n{result['result']}Stdout:\n{result['stdout']}Stderr:\n{result['stderr']}"
+        except HTTPStatusError as e:
+            error_message = e.response.text if e.response.text else e.response.reason_phrase
+            raise FunctionExecutionException(
+                f"Code execution failed with status code {e.response.status_code} and error: {error_message}"
+            )
```
Comment (eavanvalkenburg) [inline python/semantic_kernel/core_plugins/sessions_python_tool/sessions_python_plugin.py:183]: should be `raise ... from e`

File: python/semantic_kernel/core_plugins/sessions_python_tool/sessions_python_plugin.py [hunk 2]
```diff
@@ -199,33 +208,33 @@ async def upload_file(
...
+            with open(local_file_path, "rb") as data:
+                files = {"file": (remote_file_path, data, "application/octet-stream")}
+                response = await self.http_client.post(url=url, files=files)
+                response.raise_for_status()
+                response_json = response.json()
+                return SessionsRemoteFileMetadata.from_dict(response_json["value"][0]["properties"])
+        except HTTPStatusError as e:
+            error_message = e.response.text if e.response.text else e.response.reason_phrase
+            raise FunctionExecutionException(
+                f"Upload failed with status code {e.response.status_code} and error: {error_message}"
```
Comment (eavanvalkenburg) [inline python/semantic_kernel/core_plugins/sessions_python_tool/sessions_python_plugin.py:235]: same as the other one

File: python/semantic_kernel/core_plugins/sessions_python_tool/sessions_python_plugin.py [hunk 3]
```diff
@@ -242,26 +251,36 @@ async def list_files(self) -> list[SessionsRemoteFileMetadata]:
...
+                url=url,
+            )
+            response.raise_for_status()
+            response_json = response.json()
+            return [SessionsRemoteFileMetadata.from_dict(entry["properties"]) for entry in response_json["value"]]
+        except HTTPStatusError as e:
+            error_message = e.response.text if e.response.text else e.response.reason_phrase
+            raise FunctionExecutionException(
+                f"List files failed with status code {e.response.status_code} and error: {error_message}"
+            )
```
Comment (eavanvalkenburg) [inline python/semantic_kernel/core_plugins/sessions_python_tool/sessions_python_plugin.py:270]: here as well

File: python/semantic_kernel/core_plugins/sessions_python_tool/sessions_python_plugin.py [hunk 4]
```diff
@@ -275,19 +294,24 @@ async def download_file(self, *, remote_file_path: str, local_file_path: str | N
...
+                with open(local_file_path, "wb") as f:
+                    f.write(response.content)
+                return None
+
+            return BytesIO(response.content)
+        except HTTPStatusError as e:
+            error_message = e.response.text if e.response.text else e.response.reason_phrase
+            raise FunctionExecutionException(
+                f"Download failed with status code {e.response.status_code} and error: {error_message}"
+            )
```
Comment (eavanvalkenburg) [inline python/semantic_kernel/core_plugins/sessions_python_tool/sessions_python_plugin.py:317]: One more :D

**Review by eavanvalkenburg — COMMENTED** [review]
couple of missing raise froms, otherwise good to go

File: python/semantic_kernel/core_plugins/sessions_python_tool/sessions_python_plugin.py [hunk 5]
```diff
@@ -199,32 +212,32 @@ async def upload_file(
...
-            response = await self.http_client.post(
-                url=url,
-                json={},
-                files=files,  # type: ignore
-            )
+        auth_token = await self._ensure_auth_token()
+        self.http_client.headers.update(
+            {
+                "Authorization": f"Bearer {auth_token}",
+                USER_AGENT: "SESSIONS_USER_AGENT",
```
Comment (juliomenendez) [inline python/semantic_kernel/core_plugins/sessions_python_tool/sessions_python_plugin.py:219]: Is the value supposed to be the string `"SESSIONS_USER_AGENT"` or the variable `SESSIONS_USER_AGENT`?

**Review by juliomenendez — APPROVED** [review]

File: python/semantic_kernel/core_plugins/sessions_python_tool/sessions_python_plugin.py (hunk 5 above)
Comment (moonbox3) [reply python/semantic_kernel/core_plugins/sessions_python_tool/sessions_python_plugin.py:219]: Great catch!! Not sure how that got "string-ified" :)

**Review by eavanvalkenburg — APPROVED** [review]

**Pull request merged**